core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -323,6 +326,7 @@ private[spark] class TaskSchedulerImpl(
       maxLocality: TaskLocality,
       shuffledOffers: Seq[WorkerOffer],
       availableCpus: Array[Int],
+      availableGpuIndices: Array[ArrayBuffer[String]],
```

Comment:
I think its confusing that we call them "Indices" but they're strings, not ints.  But then again, you also have code to check that they can parse as ints.    Can this be made consistent?  I think its preferable if we really know they are ints and can change all of the types from String to Int, and then use `gpuIndices` consistently.  But if that is not the case, also OK to instead use `gpuAddresses` consistently.